### PR TITLE
#8570 fallback for Attribute Table

### PR DESCRIPTION
**Pull request by belom88**
## Description
Fallback component with Spinner is done.

**Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [X] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

#8570 

**What is the new behavior?**
Spinner appears during the bundle loading

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information

No

### CODE REVIEW

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import ReactDock from 'react-dock';
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:12]: This fallback component is imported in a plugin where we are trying to reduce the entry point bundle size as much as possible so I would avoid to add a library for a fallback component.
Could we use only a div with a centered spinner?

please remove also other related functions for the dock

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { getDockSize } from '../../../selectors/featuregrid';
+import { mapLayoutValuesSelector } from '../../../selectors/maplayout';
+import BasicSpinner from '../../misc/spinners/BasicSpinner/BasicSpinner';
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:15]: Please use this component instead https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/misc/Loader.jsx

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+)(ReactDock);
+
+const FeatureEditorFallback = (props) => {
+    const dockProps = getDockProps(props.open);
+    return (
+        <Dock {...dockProps} onSizeChange={size => { props.onSizeChange(size, dockProps); }}>
+            <div className="feature-editor-fallback-container">
+                <BasicSpinner sSize="sp-big" />
+            </div>
+        </Dock>
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:46]: we don't need the resize logic for a fallback component but only a div with the same size/position of the panel and a loading spinner in the center

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    );
+};
+
+export default connect(
+    createSelector([state => state?.featuregrid?.open], (open) => ({ open }))
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:51]: this is not needed because this component is already mounted on open by the Suspense HOC

File: web/client/plugins/featuregrid/FeatureEditor.jsx [hunk 5]
```diff
@@ -24,6 +24,7 @@ import {modeSelector, changesSelector, newFeaturesSelector, hasChangesSelector,
 
 import {getPanels, getHeader, getFooter, getDialogs, getEmptyRowsView, getFilterRenderers} from './panels/index';
```
Comment (allyoucanmap) [inline web/client/plugins/featuregrid/FeatureEditor.jsx:25]: Please revert this change the Fallback component must be isolated and as much simple as possible and not be included inside the main component

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx (hunk 1 above)
Comment (belom88) [reply web/client/components/data/featuregrid/FeatureEditorFallback.jsx]: Done

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx (hunk 2 above)
Comment (belom88) [reply web/client/components/data/featuregrid/FeatureEditorFallback.jsx]: Done

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx (hunk 4 above)
Comment (belom88) [reply web/client/components/data/featuregrid/FeatureEditorFallback.jsx]: Done

File: web/client/plugins/featuregrid/FeatureEditor.jsx (hunk 5 above)
Comment (belom88) [reply web/client/plugins/featuregrid/FeatureEditor.jsx]: Done

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useMemo } from 'react';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+import { getDockSize } from '../../../selectors/featuregrid';
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:12]: This selector is including a lot of dependencies for this specific case I would suggest to write here directly the selector and remove this import

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }, [size, dockStyle]);
+    return (
+        <div className="feature-editor-fallback-container" style={containerStyle}>
+            <Loader size={100} style={{ margin: '0 auto' }} />
+        </div>
+    );
+};
+
+export default connect(createSelector(
+    getDockSize,
```
Comment (allyoucanmap) [inline web/client/components/data/featuregrid/FeatureEditorFallback.jsx:28]: this is an exception usually I would ask to use the selector but in this way it should load less dependencies in this file.
```
state => state?.featuregrid?.dockSize
```

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx (hunk 6 above)
Comment (belom88) [reply web/client/components/data/featuregrid/FeatureEditorFallback.jsx]: Done

File: web/client/components/data/featuregrid/FeatureEditorFallback.jsx (hunk 7 above)
Comment (belom88) [reply web/client/components/data/featuregrid/FeatureEditorFallback.jsx]: Done

**Review by allyoucanmap — APPROVED** [review]

**Comment by allyoucanmap** [comment]
@ElenaGallo please test on dev and let us know when we can backport on the stable branch 2022.02.xx

**Comment by tdipisa** [comment]
@allyoucanmap @ElenaGallo let's wait for the release 2022.02.00 before backporting this as soon as tests are passed.

**Comment by ElenaGallo** [comment]
Test passed on DEV, @belom88 please backport to 2022.02.xx. Thnaks
